Using mvn package -Pnative instead of -Dnative, leads to CCE

**Describe the bug**
When executing `./mvnw package -Pnative` against a newly generated project, instead of a getting a native image built, we get the following error:

```
org.junit.jupiter.api.extension.TestInstantiationException: TestInstanceFactory [io.quarkus.test.junit.QuarkusTestExtension] failed to instantiate test class [org.acme.GreetingResourceTest]: java.lang.String cannot be cast to [Ljava.lang.String;
Caused by: java.lang.ClassCastException: java.lang.String cannot be cast to [Ljava.lang.String;
```

This only happens with `master`, Quarkus `1.1.0.Final` works fine.

The issue was originally found by @aureamunoz when she was working on some Quarkus material.

There is a workaround which is to use `-Dnative`, but we should fix this nonetheless since using profiles is currently broken.

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/options/BootstrapMavenOptionsParser.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/options/BootstrapMavenOptionsParser.java
@@ -20,7 +20,7 @@ public class BootstrapMavenOptionsParser {
         final Map<String, Object> map = new HashMap<>();
         put(cmdLine, map, CLIManager.ALTERNATE_USER_SETTINGS);
         put(cmdLine, map, CLIManager.ALTERNATE_GLOBAL_SETTINGS);
-        put(cmdLine, map, CLIManager.ACTIVATE_PROFILES);
+        put(map, String.valueOf(CLIManager.ACTIVATE_PROFILES), cmdLine.getOptionValues(CLIManager.ACTIVATE_PROFILES));
 
         putBoolean(cmdLine, map, CLIManager.SUPRESS_SNAPSHOT_UPDATES);
         putBoolean(cmdLine, map, CLIManager.UPDATE_SNAPSHOTS);

